src/InstrumentationEngine/Instruction.cpp

```diff
@@ -1154,20 +995,17 @@ HRESULT MicrosoftInstrumentationEngine::CBranchInstruction::GetTargetOffset(_Out
     {
         *pOffset = m_decodedTargetOffset;
     }
-
-    CLogging::LogMessage(_T("End CBranchInstruction::GetTargetOffset"));
-
     return hr;
 }
 
 HRESULT MicrosoftInstrumentationEngine::CBranchInstruction::SetBranchTarget(_In_ IInstruction* pInstruction)
 {
     HRESULT hr = S_OK;
-
-    CLogging::LogMessage(_T("Starting CBranchInstruction::SetBranchTarget"));
     IfNullRetPointer(pInstruction);
 
+    CComPtr<CInstruction> pOldInstruction = m_pBranchTarget;
     m_pBranchTarget = (CInstruction*)pInstruction;
+    IfFailRet(CBranchTargetInfo::SetBranchTarget(this, (CInstruction*)pInstruction, pOldInstruction));
 
     if (m_pBranchTarget == nullptr)
```

Comment:
We seem to be ok with a null pInstruction, but SetBranchTarget will return E_FAIL in that situation.